pkg/virt-controller/services/template.go

```diff
@@ -264,6 +265,30 @@ func (t *templateService) RenderLaunchManifest(vmi *v1.VirtualMachineInstance) (
 		})
 	}
 
+	// Handle CPU pinning
+	if vmi.IsCPUDedicated() {
+		// schedule only on nodes with a running cpu manager
+		if vmi.Spec.NodeSelector == nil {
+			vmi.Spec.NodeSelector = make(map[string]string)
+		}
+		vmi.Spec.NodeSelector[v1.CPUManager] = "true"
```

Comment:
@vladikr this is probably what the validation code is complaining about.  This needs to be moved to the defaulting code in the new mutation webhook. 

Also, I'm not sure it's safe to mutate the vmi in this "render" function. I think the vmi pointer might be directly from the informer cache

Reply:
Yeah, thanks a lot! I've seen that this is happening because I'm updating the node selector but wasn't sure how does the mutation webhook work and where do I need to update this...

Thanks, I'll move there.

Reply:
ok, that should not be necessary. Simply add the CPUManager label to the pod only. This would be just one of  more cases which we have, where we are helping the scheduler to make some requests which we made by our VMI api (in this  case, we want pinning) schedulable. Or do we think that this needs to have that much transparency?

Reply:
At some point, I thought that it's better to show it, but you're right.
Updated.